incidents/incident_script.py

```diff
@@ -0,0 +1,124 @@
+
+# Source code based off of
+# https://github.com/googleinterns/cloud-monitoring-notification-delivery-integration-sample-code/blob/server-setup/main.py
+
+"""Use custom metric to trigger and resolve incidents.
+
+This program can perform 3 operations depending on
+the argument that is given: create a new custom metric
+called "testing_metric" in the current GCP project,
+trigger an incident, or resolve the incident. To use
+it correctly, first create the custom metric (only
+do this once). Then in the GCP website create a policy
+where the metric is "testing_metric" and the resource
+type is "GCE VM Instance", and set it to trigger when the
+most recent value of any time series is above 3.0 (only create
+the policy once). Once the custom metric and policy are
+created, the program can properly trigger and resolve an
+incident on command. Note, that if you trigger an incident,
+you must resolve the incident before you can trigger it again.
+
+
+  How to use:
+
+  $ python3 incident_script.py -h
+  $ python3 incident_script.py create-custom-metric
+  $ python3 incident_script.py trigger-incident
+  $ python3 incident_script.py resolve-incident
+"""
+
+import argparse
+import os
+import time
+
+from google.cloud import monitoring_v3
+
+
+PROJECT_ID = os.environ['GOOGLE_CLOUD_PROJECT']
+METRIC_NAME = 'testing_metric'
+TRIGGER_VALUE = 3.0
+
+
+def create_custom_metric(project_id, metric_name):
+    # [START monitoring_create_metric]
+    client = monitoring_v3.MetricServiceClient()
+    project_name = client.project_path(project_id)
+    descriptor = monitoring_v3.types.MetricDescriptor()
+    descriptor.type = 'custom.googleapis.com/' + metric_name
+    descriptor.metric_kind = (
+        monitoring_v3.enums.MetricDescriptor.MetricKind.GAUGE)
+    descriptor.value_type = (
+        monitoring_v3.enums.MetricDescriptor.ValueType.DOUBLE)
+    descriptor.description = 'A custom metric meant for testing purposes'
+    descriptor = client.create_metric_descriptor(project_name, descriptor)
+    print('Created {}.'.format(descriptor.name))
+    # [END monitoring_create_metric]
+
+
+def append_to_time_series(project_id, metric_name, point_value):
+    # [START monitoring_write_timeseries]
+    client = monitoring_v3.MetricServiceClient()
+    project_name = client.project_path(project_id)
+
+    series = monitoring_v3.types.TimeSeries()
+    series.metric.type = 'custom.googleapis.com/' + metric_name
+    series.resource.type = 'gce_instance'
+    series.resource.labels['instance_id'] = '1234567890123456789'
+    series.resource.labels['zone'] = 'us-central1-f'
+    point = series.points.add()
+    point.value.double_value = point_value
+    now = time.time()
+    point.interval.end_time.seconds = int(now)
+    point.interval.end_time.nanos = int(
+        (now - point.interval.end_time.seconds) * 10**9)
+    client.create_time_series(project_name, [series])
+    # [END monitoring_write_timeseries]
+
+
+if __name__ == '__main__':
```

Comment:
Please put this in a "main" function, and call it from this if:
http://google.github.io/styleguide/pyguide.html#317-main

This will make it easier to write unit tests on main(maybe not for this particular file, but good to get in the habit of it).

Reply:
Sounds good, I added in this change.